### ISSUE TITLE
complex-paper-datatable-card.html demo `<dom-repeat>` filter issues

### ISSUE THREAD

**Issue by David-Mulder** [David-Mulder/paper-datatable#34]
See https://github.com/Polymer/polymer/issues/3094

Can be temporarily fixed by changing line 4195 in Polymer.html to

```
keys = keys.filter(function (a, i, keys) {
  return self._filterFn(c.getItem(a), i, keys);
```

**Comment by moderndeveloperllc** [comment]
So I gather that this needs to be fixed before you implement a filter option in the component?

**Comment by David-Mulder** [comment]
@moderndeveloperllc: no, this has nothing to do with a generic filter option. Do feel free to open an issue, but it runs down to this:
- in the card you can already filter in any way you like (just trigger `retrieveVisibleData` after changing your filter)
- in the plain datatable you would need to filter yourself which could maybe be neat to add.

**Comment by mercmobily** [comment]
The Polymer developers seem very unwilling to add the array index to the callback for performance reasons. What are the actual consequence to this? I am failing to see the issue! (although it might be completely 100% clear...)
